vermouth/tests/test_molecule.py

```diff
@@ -881,16 +881,46 @@ def test_link_parameter_effector_diff_class(left_class, right_class):
 
 
 @st.composite
-def attribute_dict(draw, min_size=0, max_size=None):
+def attribute_dict(draw, min_size=0, max_size=None, max_depth=1, _recursive_depth=0):
+    """
+    Strategy that builds an attribute dictionary for meta or atoms.
+
+    Parmeters
+    ---------
+    draw:
+        Internal for hypothesis.
+    min_size: int
+        The minimum number of key-value pairs.
+    max_size: int, optional
+        The maximum number of key-value pairs. If set to `None` (default),
+        there is no bound set for the maximum size of the dictionary in the
+        same manner as :func:`hypothesis.strategies.dictionaries`.
+    max_depth: int
+        The strategy is recursive so that the values can be attribute
+        dictionaries or lists containing attribute dictionaries. This argument
+        sets the maximum depth of the recursion.
+
+    Returns
+    -------
+    hypothesis.searchstrategy.lazy.LazyStrategy
+    """
     keys = st.one_of(st.text(), st.integers(), st.none())
-    bases = (st.text(), st.integers(), st.floats(), st.none())
-    lists = st.lists(st.one_of(*bases, attribute_dict(max_size=1)), max_size=3)
-    values = st.one_of(*bases, lists, attribute_dict(max_size=1))
+    bases = [st.text(), st.integers(), st.floats(), st.none()]
+    if _recursive_depth < max_depth:
+        bases.append(attribute_dict(max_size=1, _recursive_depth=_recursive_depth + 1))
```

Comment:
You can probably get rid of the argument `_recursive_depth` by reducing `max_depth` in the recursive calls?

Reply:
Very good idea.